transport/src/main/java/io/netty/channel/Channels.java

```diff
@@ -0,0 +1,223 @@
+     * class.
+     *
+     * @param groupClass event loop group class.
+     * @return channel class if found, otherwise <code>null</code>.
+     * @see io.netty.channel.socket.DatagramChannel
+     */
+    public static Class<DatagramChannel> getDatagramChannel(Class<? extends EventLoopGroup> groupClass) {
+        return (Class<DatagramChannel>) getFromMap(datagram, groupClass);
+    }
+
+    private static Class<? extends Channel> getFromMap(
+            Map<Class<? extends EventLoopGroup>, Class<? extends Channel>> map,
+            Class<? extends EventLoopGroup> groupClass) {
+
+        if (groupClass == null) {
+            return null;
+        }
+
+        Class<? extends Channel> clazz = map.get(groupClass);
+        if (clazz == null) {
+            for (Map.Entry<Class<? extends EventLoopGroup>, Class<? extends Channel>> e : map.entrySet()) {
+                if (e.getKey().isAssignableFrom(groupClass)) {
+                    clazz = e.getValue();
+                    break;
+                }
+            }
+        }
+
+        return clazz;
+    }
+
+    private static Map<Class<? extends EventLoopGroup>, Class<? extends Channel>> createMap() {
+        return new HashMap<Class<? extends EventLoopGroup>, Class<? extends Channel>>(4);
+    }
+
+    private static Map<Class<? extends EventLoopGroup>, Class<? extends Channel>> createDatagram() {
+        Map<Class<? extends EventLoopGroup>, Class<? extends Channel>> res = createMap();
+
+        res.put(NioEventLoopGroup.class, NioDatagramChannel.class);
+        res.put(OioEventLoopGroup.class, OioDatagramChannel.class);
+        addNativeImplementations(res, "DatagramChannel");
+
+        return Collections.unmodifiableMap(res);
+    }
+
+    private static Map<Class<? extends EventLoopGroup>, Class<? extends Channel>> createClient() {
+        Map<Class<? extends EventLoopGroup>, Class<? extends Channel>> res = createMap();
+
+        res.put(NioEventLoopGroup.class, NioSocketChannel.class);
+        res.put(OioEventLoopGroup.class, OioSocketChannel.class);
+        addNativeImplementations(res, "SocketChannel");
+
+        return Collections.unmodifiableMap(res);
+    }
+
+    private static Map<Class<? extends EventLoopGroup>, Class<? extends Channel>> createServer() {
+        Map<Class<? extends EventLoopGroup>, Class<? extends Channel>> res = createMap();
+
+        res.put(NioEventLoopGroup.class, NioServerSocketChannel.class);
+        res.put(OioEventLoopGroup.class, OioServerSocketChannel.class);
+        addNativeImplementations(res, "ServerSocketChannel");
+
+        return Collections.unmodifiableMap(res);
+    }
+
+    private static void addNativeImplementations(Map<Class<? extends EventLoopGroup>, Class<? extends Channel>> map,
+                                                 String classSuffix) {
+        for (String implementation : NATIVE_IMPLEMENTATIONS) {
+            String pkg = CHANNEL_PACKAGE + "." + implementation;
```

Comment:
Because of Nio/OioEventLoopGroup.class direct reference? 

Maybe we should introduce ```EventLoopGroups``` class with the same methods with additional ```fastestInstance(int numWorkers, ThreadFactory threadFactory)``` method which would return epoll event loop group on linux if available, otherwise nio based event loop group?